test/unit/logger.test.js

```diff
@@ -0,0 +1,251 @@
+  });
+
+  it('should enable console & disable silent flag', () => {
+    process.env.LOG_TO_CONSOLE = 'true';
+
+    const newLogger = require('../../lib/logger').default;
+    const consoleTransport = newLogger.transports.find(transport => {
+      return transport.name === 'console';
+    });
+
+    expect(consoleTransport).toBeDefined();
+    expect(newLogger.silent).toEqual(false);
+  });
+
+  it('should not enable logger but still set log level', () => {
+    process.env = {};
+    process.env.ENABLE_LOGGER = 'false';
+    process.env.LOG_LEVEL = 'warn';
+
+    const newLogger = require('../../lib/logger').default;
+
+    expect(newLogger.level).toEqual('warn');
+    expect(newLogger.silent).toEqual(true);
+  });
+
+  it('should not enable DailyRotateFile when logger is disabled', () => {
+    process.env = {};
+    process.env.ENABLE_ROTATING_FILE = 'true';
+
+    const newLogger = require('../../lib/logger').default;
+    const rotateTransport = newLogger.transports.find(transport => {
+      return transport.name === 'dailyRotateFile';
+    });
+
+    expect(rotateTransport).toBeUndefined();
+  });
+});
+
+// set of tests to completely reconfigure an instance of a winston logger.
+describe('Reconfigure Logger Instance', () => {
+  xit('should reconfigure log level to verbose', () => {
+    const newLogger = logger;
+    newLogger.configure({
+      level: 'verbose',
+    });
+    const logLevel = newLogger.level;
+
+    expect(logLevel).toEqual('verbose');
+  });
+
+  xit('should reconfigure log file name', () => {
+    const newLogger = logger;
+    newLogger.clear();
+    newLogger.configure({
+      transports: [new winston.transports.File({ filename: 'combined.log' })],
+    });
+
+    const fileTransport = newLogger.transports.find(transport => {
+      return transport.name === 'file';
+    });
+    expect(fileTransport.filename).toEqual('combined.log');
+  });
+  xit('should remove console transport', () => {
+    const newLogger = logger;
+    const consoleTransport = newLogger.transports.find(transport => {
+      return transport.name === 'console';
+    });
+
+    newLogger.remove(consoleTransport);
+
+    const removedTransport = newLogger.transports.find(transport => {
+      return transport.name === 'console';
+    });
+
+    expect(removedTransport).toBeUndefined();
+  });
+
+  xit('should remove DailyRotateFile transport', () => {
+    const newLogger = logger;
+    const rotateTransport = newLogger.transports.find(transport => {
+      return transport.name === 'dailyRotateFile';
+    });
+
+    newLogger.remove(rotateTransport);
+
+    const removedTransport = newLogger.transports.find(transport => {
+      return transport.name === 'dailyRotateFile';
+    });
+
+    expect(removedTransport).toBeUndefined();
+  });
+
+  xit('should clear all transports', () => {
+    const newLogger = logger;
+    newLogger.clear();
+    const transports = newLogger.transports;
+
+    expect(transports.length).toEqual(0);
+  });
```

Comment:
Why are these skipped?

Reply:
@dpopp07 - I had originally planned on giving the user full control of the logger configuration, so I wrote tests for it. I left them in there in case we wanted to go that route in the future, but I don't have any concerns if we just get rid of them.

Reply:
Sure